src/core/Akka.Remote/Serialization/DaemonMsgCreateSerializer.cs

```diff
@@ -114,6 +114,17 @@ public override object FromBinary(byte[] bytes, Type type)
         {
             var proto = DaemonMsgCreateData.ParseFrom(bytes);
             var clazz = Type.GetType(proto.Props.Clazz);
+
+            if (clazz == null)
+            {
+                var msg =
+                    string.Format(
+                        "Could not find '{0}' in loaded assemblies. If this is a remote/cluster deploy scenario, " +
+                        "ensure the remote system has referenced the assembly containing this type.", proto.Props.Clazz);
+                    
+                throw new ActorInitializationException(msg);
```

Comment:
Two minor issues here:
1. `ActorInitializationException` feels like the wrong exception, at this stage we are just materializing the stuff needed from the wire.

2. "If this is a remote/cluster deploy" that is a given.  as the DaemonMsgCreate is only used for this very scenario. 
And "has referenced the assembly containing this type" is somewhat too strict, It would work if the assembly is available in the GAC or if the assembly simply resides in one of the directories where the host process loads assemblies.
So a more generic description of the problem would be good IMO.